resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/docker/SparkDockerImageManager.scala

```diff
@@ -97,11 +96,54 @@ private[spark] class SparkDockerImageBuilder
     buildImage("spark-integration-test-asset-server", STATIC_ASSET_SERVER_DOCKER_FILE)
   }
 
+  def deleteImages(): Unit = {
+    deleteImage("spark-driver")
+    deleteImage("spark-driver-py")
+    deleteImage("spark-executor")
+    deleteImage("spark-executor-py")
+    deleteImage("spark-shuffle")
+    deleteImage("spark-resource-staging-server")
+    deleteImage("spark-init")
+    deleteImage("spark-integration-test-asset-server")
+    deleteImage("spark-base")
+  }
+
   private def buildImage(name: String, dockerFile: String): Unit = {
-    dockerClient.build(
-      DOCKER_BUILD_PATH,
-      name,
-      dockerFile,
-      new LoggingBuildHandler())
+    log.info(s"Building Docker image - $name:$dockerTag")
+    val dockerFileWithBaseTag = new File(DOCKER_BUILD_PATH.resolve(
+        s"$dockerFile-$dockerTag").toAbsolutePath.toString)
+    dockerFileWithBaseTag.deleteOnExit()
+    try {
+      val originalDockerFileText = Files.readLines(
+          DOCKER_BUILD_PATH.resolve(dockerFile).toFile, Charsets.UTF_8).asScala
+      val dockerFileTextWithProperBaseImage = originalDockerFileText.map(
+          _.replace("FROM spark-base", s"FROM spark-base:$dockerTag"))
```

Comment:
Is there anyway we can create pointers to the docker images themselves, or at least store their `RMI` so we don't have to go into the contents of the Dockerfile?

Reply:
Not entirely sure - we're always building all of the Docker images from scratch including the `spark-base` image with the given tag, and the children images need to be built with the base set with the appropriate tag.

One option is to push this to the Maven level instead. The pre-integration-test phase can copy the Dockerfiles from docker-minimal-bundle but instead of keeping the tags, Maven can set the tags accordingly. I'm not sure what the Maven primitives would be to do this.

Reply:
What kind of tags would you want to set? Something unique about the build?

Reply:
The docker image could be tagged with the git hash of the head commit of the repository. This should be unique per build.

Reply:
Seems like the maven mojo you're looking for is the 'buildnumber-maven-plugin': http://www.mojohaus.org/buildnumber-maven-plugin/